x-pack/plugins/cases/public/components/create/form_context.tsx

```diff
@@ -89,6 +94,33 @@ export const FormContext: React.FC<Props> = ({
     return formData;
   };
 
+  const transformCustomFieldsData = useCallback(
+    (customFields: { [x: `customFields.${string}`]: string | boolean }) => {
```

Comment:
Sorry, my mistake. I was confused with how the form lib works. The correct type is: `customFields: Record<string, unknown>`. If `unknown` gives you errors you can change it to `string | boolean`.

Reply:
Also the `initialCaseValue` should set the `customFields: {}` to avoid issues with `undefined` values. If not set then the type of the `customFields` should be `customFields?: ...`. Same with the `FormProps`.